up42/processing.py

```diff
@@ -64,16 +70,26 @@ class Job:
     status: JobStatus
     created: datetime.datetime
     updated: datetime.datetime
+    collection_url: Optional[str] = None
     started: Optional[datetime.datetime] = None
     finished: Optional[datetime.datetime] = None
 
+    @property
+    def collection(self) -> Optional[pystac.Collection]:
+        if self.collection_url is None:
+            return None
+        collection_id = self.collection_url.split("/")[-1]
+        client: base.StacClient = base.StacClient()
+        return client.get_collection(collection_id)  # type: ignore
```

Comment:
type: ignore, required due to static mypy analysis is not able to infer the dynamic properties of a descriptor.